workflow/executor/resource.go

```diff
@@ -77,7 +78,11 @@ func inferObjectSelfLink(obj unstructured.Unstructured) string {
 	gvk := obj.GroupVersionKind()
 	// This is the best guess we can do here and is what `kubectl` uses under the hood. Hopefully future versions of the
 	// REST client would remove the need to infer the plural name.
-	pluralGVR, _ := meta.UnsafeGuessKindToResource(gvk)
```

Comment:
Can you check if there’s any improvement to this function in the new versions of meta?

Reply:
This is the PR that is open to fix the issue, but it's not merged and it's unclear to me if people are happy with this solution or if they want to use the solution I did in this PR. 

https://github.com/kubernetes/kubernetes/pull/110053

In master it's still in silly mode https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L143